my_sprites.py

```diff
@@ -1,4 +1,83 @@
 import arcade
+import math
+
+class Enemy(arcade.Sprite):
+    """
+    parent class for all enemies in the game
+    """
+
+    def __init__(self, filename, center_x, center_y, max_hp, speed, scale=1.0):
+
+        super().__init__(
+            filename=filename,
+            scale=scale,
+            center_x=center_y,
+            center_y=center_x
+        )
+
+        # hp
+        self.max_hp = max_hp
+        self.cur_hp = max_hp
+
+        self.speed = speed
+
+        # pathfinding
+        self.path = None
+        self.cur_position = 0  # which point on the path we are heading for.
+
+    def find_path(self, barrier_list: arcade.AStarBarrierList, target_pos: tuple[int, int]):
+        """
+        calculates a path to the target pos. Sets the sprite's path to this path.
+        If an enemy has a path, it will automatically follow it.
+        """
+
+        # calculate the path. It will be a list of positions(lists)
+        self.path = arcade.astar_calculate_path(self.position,
+                                                target_pos,
+                                                barrier_list,
+                                                diagonal_movement=True)
+
+        # reset this because we are at the start of a new path
+        self.cur_position = 0
+
+    def on_update(self, delta_time: float = 1 / 60):
+
+        # follow the path, if present
+        if self.path:
+
+            # next position to move to
+            dest_x = self.path[self.cur_position][0]
```

Comment:
You can assign these in one line:
```
dest_x, dest_y = self.path[self.cur_position]
```

Maybe you could have a _dest_ that was a position? tuple[int, int]?